### ISSUE TITLE
IDE0052: 'Remove unread private members' should not be reported for private record fields

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#77907]
Although field `_id` is private and is not used explicitly, it is used by the generated GetHashCode and Equals methods.

```c#
internal readonly record struct C
{
    private readonly int _id;

    private C(int id)
        => _id = id;
}
```

**Comment by CyrusNajmabadi** [comment]
This is a bit of a challenging space.  It doesn't seem common to have record structs that contain data purely for equality/hashing purposes, but which otherwise are unused/exported from the type.  If we disable on all private fields/props of a struct, we risk truly not telling the user about cases where they genuinely have data they were previously using, but now have no need for, and would really want to remove.

this may be a case where the legitimate answer is to suppress the diagnostic, with a comment explaining the exact purpose/intent here.

**Comment by CyrusNajmabadi** [comment]
Closing as by design.  It seems more likely than not that this is actually not intentional and the field should be removed.  In the case where that's not so, and the data is private and truly only used for equals/gethashcode, the supression is appropriate.
